### PR TITLE
feat: add native Skeleton Loader to UI library

### PR DESCRIPTION
**Pull request by pete-watters**
This PR is a clone of [this one](https://github.com/leather-io/mono/pull/411) from an external contributor. 

Thanks @adrianocola 👏

### CODE REVIEW

File: packages/ui/src/components/skeleton-loader/skeleton-loader.native.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { LinearGradient } from 'expo-linear-gradient';
+
+import { Box } from '../box/box.native';
+
+const AnimatedLinearGradient = Animated.createAnimatedComponent(LinearGradient);
+
+export const ANIMATION_DURATION = 1_000;
+const ANIMATION_EASING = Easing.ease;
+const LEFT_POS_START = -100;
+const LEFT_POS_END = 100;
```
Comment (fbwoolf) [inline packages/ui/src/components/skeleton-loader/skeleton-loader.native.tsx:23]: Is there a reason these are all caps? Typically in the extension we only use this when constants are shared globally. Should these be?

**Review by fbwoolf — APPROVED** [review]

File: packages/ui/src/components/skeleton-loader/skeleton-loader.native.tsx (hunk 1 above)
Comment (pete-watters) [reply packages/ui/src/components/skeleton-loader/skeleton-loader.native.tsx:23]: **NO REASON** 😄 

I can update, thanks

Comment (kyranjamie) [reply packages/ui/src/components/skeleton-loader/skeleton-loader.native.tsx:23]: We have a handful of conventions mentioned in Notion https://www.notion.so/trustmachines/Naming-constants-1a66b44cd8234f11aadaadff5e45a3f1?pvs=4

Would love to grow this and add more

Comment (adrianocola) [reply packages/ui/src/components/skeleton-loader/skeleton-loader.native.tsx:23]: @kyranjamie, can these conventions and standards be made public? I need to request access to view them.

Comment (pete-watters) [reply packages/ui/src/components/skeleton-loader/skeleton-loader.native.tsx:23]: @adrianocola notion is not public but here's what it says: 
Screenshot 2024-09-05 at 06 08 45

I will rename them here so we can merge more efficiently

**Pull request merged**